File: C++/fizzbuzz.cpp
#include <iostream>
#include <stdlib.h>

using namespace std;

typedef unsigned int u32;

static void fizzbuzz(u32 n){
  for (u32 i = 0; i <= n; ++i){
    if (i % 3 == 0 && i % 5 == 0)
      cout << "FizzBuzz ";
    else if (i % 3 == 0)
      cout << "Fizz ";
    else if (i % 5 == 0)
      cout << "Buzz ";
    else
      cout << i << " ";
  }
  cout << endl;
}

int main(void){
  
  u32 n;

  cout << "Type any number :" << endl; 
  cin >> n;
  
  fizzbuzz(n);
  return 0;
}

File: Rust/fizzbuzz.rs
use std::io;

fn fizzbuzz(n : u32) {
   for i in 1..=n {
       if i % 5 == 0 && i % 3 == 0 {
           print!("FizzBuzz ");
       }
       else if i % 3 == 0 {
           print!("Fizz ");
       }
       else if i % 5 == 0 {
           print!("Buzz ");
       }
       else{
           print!("{} ",i);
       }  
   }
}

fn main(){
    
    println!("Type any number:");
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    
    let n: u32 = input.trim().parse().unwrap();
    println!("{:?}", fizzbuzz(n));
}
